### PR TITLE
Cmake cleanup, part 2

### PR DESCRIPTION
**Pull request by pcercuei**
This PR is for the 'dev' branch of libiio.

It bumps the minimal CMake version to 3.10, moves IIOD's install rules into IIOD's CMake script, move libiio's install rules into its own `cmake/Install.cmake` script, and update the scripts to the more modern 3.x API.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,14 +25,47 @@ endif()
...
+target_compile_definitions(iio PRIVATE LIBIIO_EXPORTS)
+
+target_compile_definitions(iio PUBLIC
+	# Request Posix 2008.09
+	_POSIX_C_SOURCE=200809L __XSI_VISIBLE=500
+
+	# Per http://www.mingw.org/wiki/Use_more_recent_defined_functions
+	$<$<PLATFORM_ID:MinGW>:_WIN32_WINNT=0x600 WINVER=0x600>
+
+	#full Single Unix Standard v3 (SUSv3) conformance (the Unix API)
```
Comment (nunojsa) [inline CMakeLists.txt:55]: nit: `# `?

**Review by nunojsa — APPROVED** [review]
lgtm

File: CMakeLists.txt (hunk 1 above)
Comment (pcercuei) [reply CMakeLists.txt:55]: Probably a bad copy-paste. I'll remove it. Thanks!

Comment (pcercuei) [reply CMakeLists.txt:55]: Actually that was already there - I'll remove it nonetheless.

**Pull request merged**